src/iidm/ThreeWindingsTransformerAdder.cpp

```diff
@@ -71,14 +89,20 @@ ThreeWindingsTransformer& ThreeWindingsTransformerAdder::add() {
 }
 
 const Network& ThreeWindingsTransformerAdder::getNetwork() const {
-    return m_substation.getNetwork();
+    if (static_cast<bool>(m_network)) {
+        return m_network.get();
+    }
+    if (static_cast<bool>(m_substation)) {
+        return m_substation.get().getNetwork();
+    }
+    throw PowsyblException("Three windings transformer has no container");
 }
 
 Network& ThreeWindingsTransformerAdder::getNetwork() {
-    return m_substation.getNetwork();
+    return const_cast<Network&>(static_cast<const ThreeWindingsTransformerAdder*>(this)->getNetwork());
```

Comment:
Notice: sonar is not happy with thoses hacks... But I think it is better than duplicating const methods content...